File: DataBaseRoomApp/jroomapp/src/main/java/com/woowrale/jroomapp/di/threads/UIThread.java
package com.woowrale.jroomapp.di.threads;

import com.woowrale.jroomapp.usecases.threads.UIScheduler;

import javax.inject.Inject;
import javax.inject.Singleton;

import io.reactivex.Scheduler;
import io.reactivex.android.schedulers.AndroidSchedulers;

/**
 * MainThread (UI Thread) implementation based on a {@link Scheduler}
 * which will execute actions on the Android UI thread.
 */
@Singleton
public class UIThread implements UIScheduler {

  @Inject
  UIThread() {}

  @Override
  public Scheduler getScheduler() {
    return AndroidSchedulers.mainThread();
  }

}

File: DataBaseRoomApp/kroomapp/src/main/java/com/woowrale/kroomapp/di/threads/UIThread.kt
package com.woowrale.kroomapp.di.threads

import com.woowrale.kroomapp.usecases.threads.UIScheduler
import io.reactivex.Scheduler
import io.reactivex.android.schedulers.AndroidSchedulers
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class UIThread @Inject internal constructor() : UIScheduler {
    override val scheduler: Scheduler
        get() = AndroidSchedulers.mainThread()
}